src/integration-test/java/com/commercetools/sync/integration/externalsource/products/ProductSyncWithNestedReferencedCategoriesIT.java

```diff
@@ -128,6 +119,13 @@ void setupTest() {
             .join();
     }
 
+    @BeforeEach
+    void setupTest() {
+        clearSyncTestCollections();
+        deleteAllProducts(CTP_TARGET_CLIENT);
+        syncOptions = buildSyncOptions();
```

Comment:
because it's not needed to be created and deleted within every single test. I moved creation and deletion (inside `deleteProductSyncTestData` in the `@BeforeAll` setup.